Automattic/woocommerce-payments
# Hide explicit currency formatting from customer-facing emails

Fixes #3513

#### Changes proposed in this Pull Request

Explicit currency formatting (e.g. with the "USD" in "$28.00 USD") should not appear in customer-facing pages when multi-currency is disabled, but currently it does show up in the order total for invoice emails sent from the order admin page.

This is because `should_output_explicit_price` currently says that explicit prices should be shown for all admin pages (when `is_admin` is true). But if an order email is sent via Order Actions on the admin page, `is_admin` is true so the explicit currency formatting ends up appearing in a customer-facing email.

This PR checks if the price is being calculated for a customer email, and handles that case appropriately in `should_output_explicit_price`.

If we can think of other cases where `is_admin` is true but we're generating customer-facing currency strings, those should probably be handled similarly.

#### Testing instructions

1. In Payments > Settings > Advanced Settings, make sure **Enable Multi-Currency** is unchecked.
2. In WooCommerce > Orders, click on any order in your store's default currency.
3. Under Order Actions, choose "Email invoice / order details to customer" and click Update.
4. Check the email that was sent (e.g. with [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/)) and verify that the "Total:" field does not have explicit currency formatting (e.g. "USD") anymore.

![Before and After](https://user-images.githubusercontent.com/16066224/146454676-fb56ca16-b478-465b-86b2-29a019e89364.png)

-------------------


**Post merge**
